### PR TITLE
Provide a list of all tunable parameters

### PR DESCRIPTION
**Pull request by orduno**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #951 |
| Primary OS tested on | Ubuntu 18.04 |
| Robotic platform tested on | Gazebo simulation of TB3 |

---

## Description of contribution in a few bullet points

Addressing #951: Adding to the documentation site a list of all tunable nav2 parameters.

### CODE REVIEW

File: sphinx_doc/howtos/tunning/params/tunable-params.rst [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+Hosted on the `PlannerServer` node
+
+.. code-block:: yaml
+
+    local_costmap:
+      ros__parameters:
+        always_send_full_costmap: true
+        clearable_layers:
+        - obstacle_layer
+        footprint: '[]'
```
Comment (SteveMacenski) [inline sphinx_doc/howtos/tunning/params/tunable-params.rst]: Why's footprint `'[]'`?

File: sphinx_doc/howtos/tunning/params/tunable-params.rst [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+        footprint_padding: 0.009999999776482582
```
Comment (SteveMacenski) [inline sphinx_doc/howtos/tunning/params/tunable-params.rst]: padding should probably be `0.01`

File: sphinx_doc/howtos/tunning/params/tunable-params.rst [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+            min_obstacle_height: 0.0
+            observation_persistence: 0.0
+            obstacle_range: 2.5
+            raytrace_range: 3.0
+            sensor_frame: ''
+            topic: /scan
+        origin_x: 0.0
+        origin_y: 0.0
+        plugin_names:
+        - obstacle_layer
```
Comment (SteveMacenski) [inline sphinx_doc/howtos/tunning/params/tunable-params.rst]: Can you include voxel layer here too with the parameters it has?

File: sphinx_doc/howtos/tunning/params/tunable-params.rst [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+.. rst-class:: content-collapse
+
+Global Costmap
+=================
+
+Hosted on the `ControllerServer` node
+
+.. code-block:: yaml
+
+    global_costmap:
```
Comment (SteveMacenski) [inline sphinx_doc/howtos/tunning/params/tunable-params.rst]: Mirrored comments from local

**Comment by orduno** [comment]
> I also notice none of the non-default plugin parameters Navfn, voxel layer are included. I see some navfn parameters but under the planner server namespace

Since we're passing these as `ros__parameters` these would be under the corresponding node, in this case `PlannerServer` is the node.

**Comment by SteveMacenski** [comment]
@orduno I think grouping them by package though makes more sense for documentation which I thought was kind of the point of this

File: sphinx_doc/howtos/tunning/index.rst [hunk 5]
```diff
@@ -0,0 +1,12 @@
+.. _docs_tunning:
+
+Tunning Guidelines
+##################
+
+This guide provides a process through which the user can adjust the tunable parameters to obtain
```
Comment (bpwilcox) [inline sphinx_doc/howtos/tunning/index.rst]: This is not (for now) so much a "guide" as it is a list of all parameters. Is the intention to provide a more fleshed out tuning guide here?

Comment (orduno) [reply sphinx_doc/howtos/tunning/index.rst]: Yes, I'm thinking this is a placeholder where we'll place links to tunning different components of the stack -- but for now, we only provide a list of the parameters.

**Comment by orduno** [comment]
Waiting on #1291 to get merged and update with voxel layer params.

**Comment by SteveMacenski** [comment]
#1291 has been merged, is this GTG with the updated voxel params?

**Pull request merged**